pages/gaming/[id].js

```diff
@@ -0,0 +1,460 @@
+                  </RegularButton>
+                )}
+
+                {isCancel && (
+                  <>
+                    <p>Are you sure you want to cancel the game?</p>
+                    <ButtonWrapper>
+                      <RegularButton
+                        type="button"
+                        onClick={() => {
+                          router.push("/gaming/");
+                        }}
+                        variant="warning"
+                      >
+                        Yes
+                      </RegularButton>
+                      <RegularButton
+                        type="button"
+                        onClick={toggleCancel}
+                        variant="confirm"
+                      >
+                        No
+                      </RegularButton>
+                    </ButtonWrapper>
+                  </>
+                )}
+              </StyledCancelContainer>
+
+              {selectedCardId && (
+                <StyledOutgreyContainer onClick={() => setSelectedCardId(null)}>
+                  <StyledLargeFlashcard $collectionColor={collectionColor}>
+                    <StyledCloseButtonContainer>
+                      <RoundButton
+                        onClick={() => setSelectedCardId(null)}
+                        type="button"
+                        variant="edit"
+                      >
+                        <MarkAsIncorrect />
+                      </RoundButton>
+                    </StyledCloseButtonContainer>
+                    <StyledSelectedCardContent>
+                      {selectedCard.content}
+                    </StyledSelectedCardContent>
+                  </StyledLargeFlashcard>
+                </StyledOutgreyContainer>
+              )}
+
+              {showOverlay && (
+                <StyledOverlay
+                  onClick={() => handleOverlayClick(flippedCards)}
+                />
+              )}
+            </>
+          )}
+
+          {isGameWon && (
+            <StyledWinningSection>
+              <StyledSuccessMessage>
+                YAY, YOU WON! <span aria-label="party-popper-emoji">🎉</span>
+              </StyledSuccessMessage>
+
+              <RegularButton
+                type="button"
+                onClick={startNewGame}
+                variant="confirm"
+              >
+                New Game
+              </RegularButton>
+              <RegularButton
+                type="button"
+                onClick={() => {
+                  router.push("/");
+                }}
+                variant="confirm"
+              >
+                Home
+              </RegularButton>
+            </StyledWinningSection>
+          )}
+        </>
+      )}
+    </>
+  );
+}
+
+const StyledFlashcardMemoryGrid = styled.ul`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: repeat(6, 1fr);
+  grid-gap: 10px;
+  list-style: none;
+`;
+
+const StyledCardContainer = styled.li`
```

Comment:
I changed the `li` elements to `button` elements and the `ul` to a `section` (because the `ul` doesn't make sense without `li`). (And then adjusted the styling accordingly, of course.) That's okay, right?